src/services/cache/CacheService.ts

```diff
@@ -76,6 +76,21 @@ export class CacheService {
     return deferred.promise;
   }
 
+  getCacheWithConfiguration(container: ICacheContainer, type: string, name: string): ng.IPromise<ICache> {
+    let deferred: ng.IDeferred<ICache> = this.$q.defer<ICache>();
+    let request: IDmrRequest = {
+      address: this.generateContainerAddress(container.name, container.profile).concat(type, name),
+    };
+    this.dmrService.readResource(request).then((response) => {
+      let c: ICache = new Cache(name, type, response.configuration);
+      this.getCacheTemplate(container, type, name).then((response) => {
+        c.configModel = response;
+        deferred.resolve(c);
+      });
+    });
```

Comment:
We should re-use the `getCache` method

```typescript
getCacheWithConfiguration(container: ICacheContainer, type: string, name: string): ng.IPromise<ICache> {
    let deferred: ng.IDeferred<ICache> = this.$q.defer<ICache>();
    this.getCache(name, type, container.name, container.profile)
      .then(cache => {
        this.getCacheTemplate(container, type, name)
          .then(template => {
            cache.configModel = template;
            deferred.resolve(cache);
          })
      });
    return deferred.promise;
  }
```

We should have the scenarios where any of these operations fail as well

Reply:
+1 I know we should re-use getCache but it getCacheTemplate calls getCache and then I would introduce recursion. For such a small fix I would not want to risk potential problems for a PR that needs to go downstream in 7.2. We can revisit all these cache service methods post release. I'll add deferred rejecting for errors.

Reply:
Whoops, I didn't realise that getCacheTemplate called `getCache` good spot :+1: